core/app/models/spree/unit_cancel.rb

```diff
@@ -24,6 +24,9 @@ def adjust!
       eligible: true,
       finalized: true
     )
+
+    inventory_unit.line_item.order.recalculate
```

Comment:
what about manually updating the adjustment_total instead?

```ruby
inventory_unit.line_item.adjustment_total += amount
```

This would ensure it's updated for next iterations without hard refreshing all the order content

Reply:
I tried that, but the line items that are on the inventory units were all different objects that pointed to the same line item record. So each time the code came through the adjustment_total was still at 0.

Reply:
mhh, I tried this code as well and it seems to work with your spec (removing the `line_item.reload`)

Reply:
It worked for me in the spec as well, but once I tried it in the sandbox app it failed. I will try again however and see if it was just a fluke failure.